src/picongpu/include/particles/flylite/helperFields/LocalRateMatrix.hpp

```diff
@@ -0,0 +1,118 @@
+/* Copyright 2017 Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+
+// PMacc
+#include "dataManagement/ISimulationData.hpp"
+#include "dimensions/GridLayout.hpp"
+#include "memory/buffers/GridBuffer.hpp"
+#include "memory/Array.hpp"
+
+#include <string>
+#include <memory>
+
+
+namespace picongpu
+{
+namespace particles
+{
+namespace flylite
+{
+namespace helperFields
+{
+    using namespace PMacc;
+
+    class LocalRateMatrix :
+        public ISimulationData
+    {
+    private:
+        /** A[iz, numpop, numpop] */
```

Comment:
it should for an implementer that reads `private:`: it's the index order in the rate matrix, more a note to myself